pandas/core/indexes/datetimes.py

```diff
@@ -93,7 +94,14 @@ def f(self):
             result = fields.get_date_field(values, field)
             result = self._maybe_mask_results(result, convert='float64')
 
-        return Index(result, name=self.name)
+        if field in ['weekday_name', 'day_name']:
+            cats = ['Monday', 'Tuesday', 'Wednesday', 'Thursday',
```

Comment:
These are already defined in the `calendar` standard library (`calendar.day_name`) or here:  https://github.com/pandas-dev/pandas/blob/be90d492836fa604b4b914ab6c7387752a6ba9e6/pandas/_libs/tslibs/ccalendar.pyx#L50